src/composer/src/cossystem.rs

```diff
@@ -28,13 +28,20 @@ pub struct InterfaceVariant {
     pub variant: Option<String>,
 }
 
+#[derive(Debug, Deserialize,Clone)]
```

Comment:
space before Clone. Did you run rustfmt? You should never do a commit without running that on the project.